File: LC/2511.captureForts.py
class Solution:
    def captureForts(self, forts: List[int]) -> int:
        ans = 0
        for i,x in enumerate(forts):
            if x != 1:continue
            j = i - 1
            while j >= 0 and forts[j] == 0:j -= 1
            if j >= 0 and forts[j] == -1:
                ans = max(i - j - 1,ans)
            j = i + 1
            while j < len(forts) and forts[j] == 0:j += 1
            if j < len(forts) and forts[j] == -1:
                ans = max(j - i - 1,ans)
        return ans

File: LC/2511.captureForts.cpp
class Solution {
public:
    int captureForts(vector<int>& forts) {
        int n = forts.size(),ans = 0;
        for (int i = 0;i < n;++i) {
            if (forts[i] == 1) {
                int cnt = 0;
                for (int j = i - 1;j >= 0;j--) {
                    if (forts[j] == 0) cnt++;
                    else if (forts[j] == -1) {
                        ans = max(ans,cnt);
                        break;
                    } 
                    else break;
                }
                cnt = 0;
                for (int j = i + 1;j < n;j++) {
                    if (forts[j] == 0) cnt++;
                    else if (forts[j] == -1) {
                        ans = max(ans,cnt);
                        break;
                    } 
                    else break;
                }
            }
        }
        return ans;
    }
};
